src/sentry/grouping/enhancer/__init__.py

```diff
@@ -145,10 +206,60 @@ def apply_modifications_to_frame(
         extra_fingerprint: str = "",
     ) -> None:
         """This applies the frame modifications to the frames itself. This does not affect grouping."""
-        in_memory_cache: dict[str, str] = {}
 
         # Matching frames are used for matching rules
         match_frames = [create_match_frame(frame, platform) for frame in frames]
+
+        rust_enhanced_frames = None
+        try:
+            use_rust_enhancements = random.random() < options.get(
+                "grouping.rust_enhancers.modify_frames_rate"
+            )
+        except Exception:
+            use_rust_enhancements = False
+        if (enhancements := self.rust_enhancements) and use_rust_enhancements:
+            e = exception_data or {}
+            e = {
+                "ty": e.get("type"),
+                "value": e.get("value"),
+                "mechanism": get_path(e, "mechanism", "type"),
+            }
+            for key in e.keys():
+                value = e[key]
+                if isinstance(value, str):
+                    e[key] = value.encode("utf-8")
+
+            try:
+                rust_enhanced_frames = enhancements.apply_modifications_to_frames(
+                    iter(match_frames), e
+                )
+                metrics.incr("rust_enhancements.modifications_run")
+            except Exception:
+                logger.exception("failed running Rust Enhancements modifications")
+
+            # TODO: once ready, update the frame directly
+            # for frame, (category, in_app) in zip(frames, rust_enhanced_frames):
+            #     if in_app is not None:
+            #         set_in_app(frame, in_app)
+            #     if category is not None:
+            #         set_path(frame, "data", "category", value=category)
+
+        def compare_rust_enhancers():
+            if rust_enhanced_frames:
+                has_mismatch = False
+                for frame, (category, in_app) in zip(frames, rust_enhanced_frames):
+                    in_app_matches = bool(frame.get("in_app")) == in_app
+                    category_matches = get_path(frame, "data", "category") == category
+                    is_mismatch = not in_app_matches or not category_matches
+                    has_mismatch = has_mismatch or is_mismatch
+                if has_mismatch:
+                    logger.error(
+                        "Rust Enhancements mismatch",
+                        extra={"frames": frames, "rust_enhanced_frames": rust_enhanced_frames},
```

Comment:
You may find some stripping happening. In the past, I have iterated prior to the `logger.error` over the large structure and do some `logger.info` since it would show up as breadcrumbs for inspection. See [this PR](https://github.com/getsentry/sentry/commit/e7e94398ab755d008a43fd7e32fad532c1533511) for an example and followed by [this other one](https://github.com/getsentry/sentry/pull/63512).